### ISSUE TITLE
DefaultHttpRequester doesn't work with version 1 cookie

### ISSUE THREAD

**Issue by oliverwhk** [AngleSharp/AngleSharp#519]
With default Http requester, it's not compatible with version 1 cookie.

My scenario is:
1) Open page 1, response came back with version 1 cookies
2) Open page 2 (request sent with version 1 cookie) but failed

System.Net.CookieException : An error occurred when parsing the Cookie header for Uri 'http://localhost/'.
  ----> System.Net.CookieException : Cookie format error.

Following is the code to mimic my scenario:

```
var config = Configuration.Default.WithDefaultLoader().WithCookies();
            var context = BrowsingContext.New(config);
            const string cookieValue = "FGTServer=04E2E1A642B2BB49C6FE0115DE3976CB377263F3278BD6C8E2F8A24EE4DF7562F089BFAC5C0102; Version=1";

            var document = await context.OpenAsync(res => res.Content("<div></div>")
                                        .Address("http://localhost/")
                                        .Header(HeaderNames.SetCookie, cookieValue));

            await context.OpenAsync("http://localhost/");
```

**Comment by oliverwhk** [comment]
I had a look myself and think following change should fix it but I need second pair of eyes who are more experienced in this project.

```
private void SetCookies()
{
    var cookieHeader = _request.Headers.GetOrDefault(HeaderNames.Cookie, String.Empty);
    // UPDATED LINE
    _cookies.SetCookies(_http.RequestUri, cookieHeader.Replace(';', ',').Replace("$", ""));
    // ORIGINAL LINE
    //                _cookies.SetCookies(_http.RequestUri, cookieHeader.Replace(';', ','));
}
```

**Comment by FlorianRappl** [comment]
Actually, the original title was accurate (changed back). I've integrated the test and verified the proposed fix; seems to work fine (but I doubt that this an ideal solution; now special properties such as `$Version` will be read as cookie values, e.g., `Version`). But its better than nothing!

Fixed, see PR #522.

**Comment by oliverwhk** [comment]
Cool thanks
